### PR TITLE
core(network-request): simplify recomputeTimesWithResourceTiming

### PR DESCRIPTION
**Pull request by connorjclark**
Adds a comment and removes a conditional that is impossible to be true.

### CODE REVIEW

File: core/lib/network-request.js [hunk 1]
```diff
@@ -372,16 +373,22 @@ class NetworkRequest {
...
+    const responseTimestamp = this.responseHeadersEndTime;
+
+    // Update this.responseHeadersEndTime. All timing values from the netstack (timing) are well-ordered, and
+    // so are the timestamps from CDP (which this.responseHeadersEndTime belongs to). It shouldn't be possible
+    // that this timing from the netstack is greater than the onResponse timestamp, but just to ensure proper order
+    // is maintained we bound the new timing by the network request time and the response timestamp.
+    this.responseHeadersEndTime = headersReceivedTime;
+    this.responseHeadersEndTime = Math.min(this.responseHeadersEndTime, responseTimestamp);
     this.responseHeadersEndTime = Math.max(this.responseHeadersEndTime, this.networkRequestTime);
+
```
Comment (connorjclark) [inline core/lib/network-request.js:391]: `headersReceivedTime` with the value of `this.responseHeadersEndTime` after, for every run I tried (including lantern database) there is no difference. This code originally came from CDT (no explanation there) - best I can reason is that it was to be extra careful that ordering invariants are maintained. I don't think it's ever actually needed, though. Adding a comment here for better readability.

File: core/lib/network-request.js [hunk 2]
```diff
@@ -372,16 +373,22 @@ class NetworkRequest {
...
     // If we don't have receiveHeadersEnd, we really don't have more accurate data.
     if (timing.requestTime === 0 || timing.receiveHeadersEnd === -1) return;
+
     // Take networkRequestTime and responseHeadersEndTime from timing data for better accuracy.
     // Timing's requestTime is a baseline in seconds, rest of the numbers there are ticks in millis.
     this.networkRequestTime = timing.requestTime * 1000;
     const headersReceivedTime = this.networkRequestTime + timing.receiveHeadersEnd;
-    if (!this.responseHeadersEndTime || this.responseHeadersEndTime < 0) {
-      this.responseHeadersEndTime = headersReceivedTime;
-    }
```
Comment (connorjclark) [inline core/lib/network-request.js:381]: this literally cannot be true, the one place this is called sets `responseHeadersEndTime` to a non-zero value first.

File: core/lib/network-request.js [hunk 3]
```diff
@@ -372,16 +373,22 @@ class NetworkRequest {
     // Don't recompute times if the data is invalid. RequestTime should always be a thread timestamp.
     // If we don't have receiveHeadersEnd, we really don't have more accurate data.
     if (timing.requestTime === 0 || timing.receiveHeadersEnd === -1) return;
+
     // Take networkRequestTime and responseHeadersEndTime from timing data for better accuracy.
```
Comment (paulirish) [inline core/lib/network-request.js:377]:
```suggestion
    // networkRequestTime and responseHeadersEndTime are canonically defined in resource timing. Up till now, responseHeadersEndTime has a bogus fallback that was set in _onResponse().
```

File: core/lib/network-request.js [hunk 4]
```diff
@@ -372,16 +373,22 @@ class NetworkRequest {
...
     // Timing's requestTime is a baseline in seconds, rest of the numbers there are ticks in millis.
```
Comment (paulirish) [inline core/lib/network-request.js:378]:
```suggestion
    // Timing's requestTime is a baseline in seconds, rest of the numbers there are ticks in millis. Shout out https://raw.githubusercontent.com/GoogleChrome/lighthouse/main/docs/Network-Timings.svg
```

File: core/lib/network-request.js (hunk 1 above)
Comment (paulirish) [reply core/lib/network-request.js:391]: love it. this readability refactor makes me happy.

also... the DT logic here is still the same today fwiw. https://github.com/ChromeDevTools/devtools-frontend/blob/23cde35228250b831d70b63ab76d05ab4d193129/front_end/core/sdk/NetworkRequest.ts#L737-L754

**Review by paulirish — APPROVED** [review]

File: core/lib/network-request.js (hunk 3 above)
Comment (connorjclark) [reply core/lib/network-request.js:377]: lmk if the new comment is still lacking for u

**Review by adamraine — APPROVED** [review]
Looks good, just one quick tangent...

File: core/scripts/test-lantern.sh [hunk 5]
```diff
@@ -17,7 +17,7 @@ fi
 
 printf "Determined the following files have been touched:\n\n$CHANGED_FILES\n\n"
 
-if ! echo $CHANGED_FILES | grep -E 'dependency-graph|metrics|lantern|predictive-perf|network-recorder' > /dev/null; then
+if ! echo $CHANGED_FILES | grep -E 'dependency-graph|metrics|lantern|predictive-perf|network-recorder|network-request' > /dev/null; then
```
[the remark this answers is not in the document]
Comment (connorjclark) [reply core/scripts/test-lantern.sh:20]: smokes def. take most of the time of a run, and this is only ~2m. mostly worried about the constant downloading of this thing and how that could upset cloud storage?

**Pull request merged**